### ISSUE TITLE
support struct literal syntax

### ISSUE THREAD

**Issue by universalmind303** [GlareDB/glaredb#2112]
## Description

we support array literal syntax

```sql
SELECT ['A-Wing', 'B-Wing', 'X-Wing', 'Y-Wing'] AS starfighter_list;
```

i think we should also support struct literal syntax:

```sql
SELECT {name: 'Star Destroyer', common_misconceptions: 'Can''t in fact destroy a star'} AS star_destroyer_facts;
```

## References
https://duckdb.org/2022/05/04/friendlier-sql.html#simple-list-and-struct-creation

**Comment by tychoish** [comment]
Seems super interesting and exciting.

could also have `[pair(k,v), pair(k,v)]` as a syntax/option? 

what are the values that we're producing/reading? (like do we get structs/objects or do the structs become data frames/etc.?)

**Comment by universalmind303** [comment]
> like do we get structs/objects or do the structs become data frames/etc.?)

all structs use the native arrow `Struct` dtype

> what are the values that we're producing/reading?

I think a real example may explain the use cases a bit better


Given a json file containing some coordinates nested into an object:
```json
{"coordinates": {"latitude": 20.0, "longitude": 70.5}, "description": "...."}
{"coordinates": {"latitude": 21.0, "longitude": 70.5}, "description": "...."}
```


we can currently access the fields using dot notation
```sql
> select coordinates.latitude, coordinates.longitude from ndjson_scan('test.json');
┌───────────────────────────────────┬────────────────────────────────────┐
│ ndjson_scan.coordinates[latitude] │ ndjson_scan.coordinates[longitude] │
│                                ── │                                 ── │
│                           Float64 │                            Float64 │
╞═══════════════════════════════════╪════════════════════════════════════╡
│                              20.0 │                               70.5 │
│                              21.0 │                               70.5 │
└───────────────────────────────────┴────────────────────────────────────┘
```

but there is no way to create a dataset in this shape given our current functionality.

Ideally we could transpose a flattened dataset into a struct
for example, the same json, but without a nested object
```json
{"latitude": 20.0, "longitude": 70.5, "description": "...."}
{"latitude": 21.0, "longitude": 70.5, "description": "...."}
```


> could also have [pair(k,v), pair(k,v)] as a syntax/option?
I think some other sql dialects have similar syntax, however I feel the struct literal is more intuitive/natural 

It'd be really convenient if we supported field shorthand as well. 
```sql
select {latitude, longitude} as coordinates from ndjson_scan('test.json');
-- as opposed to `{latitude: latitude, longitude: longitude}`
```

**Comment by tychoish** [comment]
> I think some other sql dialects have similar syntax, however I feel the struct literal is more intuitive/natural

yeah, makes sense; they don't conflict with each other, and the pair thing is easier to generate. 

> It'd be really convenient if we supported field shorthand as well.

this looks great, and yeah, could/should support both.
